src/TorchSharp/DisposeScope.cs

```diff
@@ -40,7 +40,7 @@ internal DisposeScope(DisposeScopeManager disposeScopeManager)
         public IReadOnlyList<IDisposable> DisposablesView {
             get {
                 if (this._disposeScopeManager is null)
-                    throw new ObjectDisposedException("The dispose scope has been disposed.");
+                    throw new ObjectDisposedException(this.GetType().FullName);
```

Comment:
3. The parameter of the constructor of `ObjectDisposedException` is not the message, but is something called objectName.